### PR TITLE
Show loss in progress bar

### PR DESCRIPTION
**Pull request by gsganden**
This PR adds reporting of the task training losses to the training progress bar to support killing training run early if you can see that it's not getting anywhere.

The number it reports is an exponentially weighted moving average of the training loss over batches. I suggest that we use these averages in the table at the end of each epoch as well as in the progress bar. I did not change the calculation of the total training loss that is also reported in that table. I'm definitely open to other approaches.

Pull Request Checklist
 - [x] All tests in the `tests` folder pass with a local build
 - [x] Pull request includes a description of why we are doing this
 - [x] Init files import new capabilities to appropriate level of package (if applicable)
 - [x] CHANGELOG has been updated
 - [x] Version in `_version.py` has been updated
 - [ ] README has been updated (if applicable)

### CODE REVIEW

**Comment by gsganden** [comment]
Here's what it look like in action:

Uploading Screen Recording 2021-01-04 at 10.36.17 AM.mov…

**Comment by gsganden** [comment]
https://user-images.githubusercontent.com/13399445/103557582-0497d200-4e79-11eb-8eaa-1ecae0e77ff1.mov

File: octopod/learner.py [hunk 1]
```diff
@@ -9,6 +9,8 @@
                                    DEFAULT_METRIC_DICT,
                                    )
 
+SMOOTH_LOSS_ALPHA = 0.2
```
Comment (parsing-science) [inline octopod/learner.py:12]: Is there a reason not to make this a parameter to the learner? I imagine in some cases someone might want a different alpha.

File: octopod/learner.py [hunk 2]
```diff
@@ -111,12 +113,11 @@ def fit(
         headers.append('time')
         pbar.write(headers, table=True)
 
+        self.smooth_train_losses = {}
```
Comment (parsing-science) [inline octopod/learner.py:116]: Can we keep the `training_loss_dict` name? Or at least call it `smooth_training_loss_dict`? That will make it more parallel with the `val_loss_dict`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,12 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project uses [Semantic Versioning](http://semver.org/).
 
+# [2.3.0] - 2020-12-31
```
Comment (parsing-science) [inline CHANGELOG.md:6]: I would consider this to be a breaking change since we are changing the output of a some of the functions. What do you think?

**Review by parsing-science — CHANGES_REQUESTED** [review]
Thanks for taking care of this @gsganden ! Do you think it's worth leaving a comment somewhere explaining that we are displaying the smoothed loss and not just the raw value? I imagine that might not be obvious to everyone.

File: octopod/learner.py (hunk 1 above)
Comment (gsganden) [reply octopod/learner.py:12]: Good idea

File: octopod/learner.py (hunk 2 above)
Comment (gsganden) [reply octopod/learner.py:116]: Yes

File: CHANGELOG.md (hunk 3 above)
Comment (gsganden) [reply CHANGELOG.md:6]: I would say no because user code won't have to change as a result of this PR. We're only making changes to side effects that I think users can mostly not worry about.

You could say yes because the train_loss values in the tables mean something different -- they will be moving averages rather than simple averages. Doing a major version bump to emphasize that change feels like overkill to me--either way you're pretty much just paying attention to when the number stops going down--but I'm not strongly opposed to it.

Comment (parsing-science) [reply CHANGELOG.md:6]: I vote yes. If you continue training a model that was previously trained with octopod, the numbers won't match up, which might be confusing. A major version change will make it more obvious that some behavior has been altered.

Comment (gsganden) [reply CHANGELOG.md:6]: OK

**Review by parsing-science — APPROVED** [review]
Looks good! Thanks for the updated docstring; it's super informative.
